mcs/class/corlib/System.Security.Cryptography/RNGCryptoServiceProvider.cs

```diff
@@ -94,31 +98,33 @@ private void Check ()
 		private static extern bool RngOpen ();
 		
 		[MethodImplAttribute(MethodImplOptions.InternalCall)]
-		private static extern IntPtr RngInitialize (byte[] seed);
+		unsafe private static extern IntPtr RngInitialize (byte* seed, IntPtr seed_length);
 
 		[MethodImplAttribute(MethodImplOptions.InternalCall)]
-		private static extern IntPtr RngGetBytes (IntPtr handle, byte[] data);
+		unsafe private static extern IntPtr RngGetBytes (IntPtr handle, byte* data, IntPtr data_length);
```

Comment:
Pass the `data_length` as a `gint32`.

Reply:
`uintptr_t mono_array_handle_length (MonoArrayHandle arr);`
but I used intptr_r as compromise to preserve previous signedness in existing code.

Reply:
We also have:
```
#ifdef MONO_BIG_ARRAYS
typedef uint64_t mono_array_size_t;
#else
typedef uint32_t mono_array_size_t;
#endif
```
however the ifdef does not translate into the BCL interface -- we only build it once without ifdef. So you have to pick one type that works either way, with either configuration of C and with C#, so the wider type is what works. The smaller type would be silently truncated and silent truncation of integers is dangerous and not worth much risk.

Reply:
https://msdn.microsoft.com/en-us/library/system.array.length%28v=vs.110%29.aspx. `Array.Length` returns `int` which is equivalent to `gint32`.

Reply:
But what about Mono's large array feature? Worthless? Only for embedding uses that aren't interacting with C#/IL?